### PR TITLE
Add blend stacking node (Z-Stack)

### PR DESCRIPTION
**Pull request by JustNoon**
Implements #1686 as well as mean, minimum, and maximum.

I chose 2-15 images with 2 being a minimum.

### CODE REVIEW

File: backend/src/packages/chaiNNer_standard/image_utility/compositing/z_stack.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+from nodes.group import group
+from nodes.properties.inputs import EnumInput, ImageInput
+from nodes.properties.outputs import ImageOutput
+from nodes.utils.utils import ALPHABET, get_h_w_c
+
+from .. import compositing_group
+
+class Expression(Enum):
+    MEDIAN = "median"
+    MEAN = "mean"
```
Comment (RunDevelopment) [inline backend/src/packages/chaiNNer_standard/image_utility/compositing/z_stack.py:17]: Which [mean](https://en.wikipedia.org/wiki/Mean#Types_of_means) is it?

Comment (JustNoon) [reply backend/src/packages/chaiNNer_standard/image_utility/compositing/z_stack.py:17]: Arithmetic. Should be the same results as if you had blended n layers with 1/n opacity each (with most transparent at the top).

**Comment by JustNoon** [comment]
I think I've finally pleased the lint and formatting demons so I've rebased to current commit.

Any review corrections I'll add as separate commits going forward.

**Comment by RunDevelopment** [comment]
This does not work if the images aren't all the same shape.

image
image


Why don't we just require that all images need to have the same size and number of channels? Would this limit the use case for this node?

**Comment by JustNoon** [comment]
> This does not work if the images aren't all the same shape.
> 
> image image
> 
> Why don't we just require that all images need to have the same size and number of channels? Would this limit the use case for this node?

It certainly wouldn't be a problem for my use-cases but I didn't know how people would use it. It could always be expanded to handle dimensions and channels.

File: backend/src/packages/chaiNNer_standard/image_utility/compositing/z_stack.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    outputs=[ImageOutput()],
+)
+def z_stack(
+    expression: Expression,
+    *args: Union[np.ndarray, None],
+) -> np.ndarray:
+    """Align and evaluate images"""
+
+    inputs: List[Union[np.ndarray, None]] = [*args]
+    images = [x for x in inputs if x is not None]
```
Comment (RunDevelopment) [inline backend/src/packages/chaiNNer_standard/image_utility/compositing/z_stack.py:48]: Just cause it's bothering me. 
```suggestion
    *inputs: np.ndarray | None,
) -> np.ndarray:
    """Align and evaluate images"""

    images = [x for x in inputs if x is not None]
```

File: backend/src/packages/chaiNNer_standard/image_utility/compositing/z_stack.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        ImageInput("Image A"),
+        ImageInput("Image B"),
+        group("optional-list")(
+            *[
+                ImageInput(f"Image {letter}").make_optional()
+                for letter in ALPHABET[2:14]
+            ],
+        ),
+    ],
+    outputs=[ImageOutput()],
```
Comment (RunDevelopment) [inline backend/src/packages/chaiNNer_standard/image_utility/compositing/z_stack.py:39]: Okay, this is going to be just a little painful. So the type system doesn't support arrays yet, so we have to reference each input image individually...
```suggestion
    outputs=[
        ImageOutput(
            image_type="""
                def conv(i: Image | null) = match i { Image => i, _ => any };

                Input1 & Input2
                    & conv(Input3)
                    & conv(Input4)
                    & conv(Input5)
                    & conv(Input6)
                    & conv(Input7)
                    & conv(Input8)
                    & conv(Input9)
                    & conv(Input10)
                    & conv(Input11)
                    & conv(Input12)
                    & conv(Input13)
                    & conv(Input14)
            """
        ).with_never_reason(
            "All input images much have the same size and number of channels."
        ),
    ],
```

**Review by RunDevelopment — COMMENTED** [review]
Last suggestions, then gtg.

**Comment by JustNoon** [comment]
Thanks for the help. Seriously.

The parameters got that way in my fight with pywright and I didn't want to touch it anymore after we had a ceasefire.

**Review by joeyballentine — APPROVED** [review]

**Pull request merged**